### DISCUSSION TITLE
NextJS 12 (React 18) as a main codebase for J!4

### DISCUSSION THREAD

**Discussion by 1337180** [joomla/joomla-cms#37524] [General]
Hello guys, I am done learning theory about this.
So I am about to explode where to start.
So I am here and asking for an advice:
I want to make strong nextjs react javascript codebase SPA with lot of features, gadges fun stuff and more.. generally programming ideas..
I want to connect Joomla! 4 to this. Is this entirely possible, what are the limits and what are dos and donts.
Also please some resources how to integrate Joomla! 4 > cassiopea > php to this.
I never seen this before so Im feeling like a pioneer into this so maybe I can show the way by my-yours advices and help.
Thanks  for answers!

Juraj

**Comment by 1337180** [comment]
How long could this take to me ? 1 year 4-5 hours a day is acceptable. I think there are some great challenges.
Some recommended approach?
Now Im thinking about how do I load cassiopeas index.php on Single Page Application.
Please, thanks.

**Comment by drmenzelit** [comment]
Duplicate? https://github.com/joomla/joomla-cms/discussions/37189#discussion-3912358

**Reply by 1337180** [reply]
this is next JS
close the other forum
thanks for that
